src/libraries/System.IO.Compression.ZipFile/tests/ZipFile.Extract.cs

```diff
@@ -103,6 +103,22 @@ public void Windows_ZipWithInvalidFileNames_ThrowsArgumentException(string zipNa
             AssertExtensions.Throws<ArgumentException>("path", null, () => ZipFile.ExtractToDirectory(compat(zipName) + ".zip", GetTestFilePath()));
         }
 
+        /// <summary>
+        /// This test checks whether or not ZipFile.ExtractToDirectory() is capable of handling filenames
+		/// which contain invalid path characters in Windows.
+        /// </summary>
+        [Theory]
+        [InlineData("InvalidWindowsFileNameChars.zip", "Test______________________________________.txt")]
+        [InlineData("InvalidWindowsFileNameChars.zip", "Test______________________________________/TestText1______________________________________.txt")]
+        [InlineData("InvalidWindowsFileNameChars.zip", "Test/normalText.txt")]
```

Comment:
this will run the test 3 times. you can just run the test once (no InlineData) and check all three files are found.